core/trino-main/src/main/java/io/trino/testing/TestingAccessControlManager.java

```diff
@@ -144,6 +144,7 @@ public class TestingAccessControlManager
     private Predicate<SchemaTableName> deniedTables = s -> true;
     private BiPredicate<Identity, String> denyIdentityTable = IDENTITY_TABLE_TRUE;
     private BiPredicate<Identity, String> denyIdentityFunction = IDENTITY_FUNCTION_TRUE;
+    private Optional<Set<String>> allowImpersonationRoles = Optional.empty();
```

Comment:
Also this should be something that holds authorization trinity. You need to have subject, action (here it is know that it is impersonation) and entity. So you need to hold pair of who (subject) can impersonate who (entity). And subject here is an (user with his roles), entity is just user.

Reply:
Why? That would make this much more complex. Here I only store roles that allow impersonations, this is for tests only and trino doesn't even have good support for roles in general..

Reply:
Let's sync on that.

Reply:
I mean there is some thinking here in the class and you are introducing new concept of access control to this testing class which makes it harder. I would like us to reuse what we have and rewrite the test.